src/bp-activity/bp-activity-notifications.php

```diff
@@ -293,6 +293,45 @@ function bp_activity_remove_screen_notifications_single_activity_permalink( $act
 }
 add_action( 'bp_activity_screen_single_activity_permalink', 'bp_activity_remove_screen_notifications_single_activity_permalink' );
 
+/**
+ * Mark notification read when user visit sing post page permalink
+ *
+ * @since BuddyPress 2.0.0
+ */
+function bp_activity_mark_blog_post_notification_read() {
+	global $post;
+
+	if ( ! is_user_logged_in() || empty( $post->post_type ) || ! bp_is_post_type_feed_enable( $post->post_type ) || ! is_singular( $post->post_type ) ) {
+		return;
+	}
+
+	$rid        = filter_input( INPUT_GET, 'rid', FILTER_VALIDATE_INT );
+	$crid       = filter_input( INPUT_GET, 'crid', FILTER_VALIDATE_INT );
```

Comment:
@janakkaneriya007 This both variable name should be understandable and I am not sure what the variable is and what we taking from the `INPUT_GET`